### PR TITLE
Google Docs/Sheets/Presentation/Forms, OneDrive (for Office 365)

### PR DESCRIPTION
**Pull request by Tatiana5**

### CODE REVIEW

File: collection/sites/googledocs.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name: All Google Documents (disable Google Sheets before enable it!)
```
Comment (rxu) [inline collection/sites/googledocs.yml:1]: > disable Google Sheets before enable it!

I'm not sure about this as end-user will never see the comment. Will this cause some conflict or something?

File: collection/sites/googledocs.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+host: docs.google.com
+example:
+    - 'https://docs.google.com/spreadsheets/d/1f988o68HDvk335xXllJD16vxLBuRcmm3vg6U9lVaYpA'
+    - 'https://docs.google.com/spreadsheets/u/1/d/e/2PACX-1vQ7iYECa_ziPWygKBf9Og-PkQl5-gdnFRU-jP-z1FyjLfLXMAdRJN1J5Lx31NjnCq5lkRcL-T0Idh4_/pubhtml'
+    - 'https://docs.google.com/spreadsheets/d/e/2PACX-1vSbbciMU7t5dCB8auk-nyK-p-25TJjgUrzBuMCAiq_2ZDqoMSFAlZCctIC7y3TXS71ys_MRdoAU4Jqr/pubchart?oid=1729348495'
+extract: "!docs\\.google\\.com/(?'id'(document|spreadsheets|presentation|forms)(((/\\d)+|(/\\w)+)+)/([^/?]+)(/[^/]+)?)!"
+iframe:
+    height: 500
+    width: '100%'
```
Comment (rxu) [inline collection/sites/googledocs.yml:10]: It should be consistent, so quoted both of `width` and `height` values for example.

File: collection/sites/googledocs.yml (hunk 1 above)
Comment (Tatiana5) [reply collection/sites/googledocs.yml:1]: The user will see the comment in the list of sites.
So I realized that the conflict is due to the same host, docs.google.com. Without this host support for documents/presentations/forms cannot be added

File: collection/sites/googledocs.yml (hunk 2 above)
Comment (Tatiana5) [reply collection/sites/googledocs.yml]: Fixed

File: collection/sites/googledocs.yml (hunk 1 above)
Comment (rxu) [reply collection/sites/googledocs.yml:1]: Okay, tried and it doesn't seem to break the board.

File: collection/sites/googledocs.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+name: All Google Documents (disable Google Sheets before enable it!)
+host: docs.google.com
+example:
+    - 'https://docs.google.com/spreadsheets/d/1f988o68HDvk335xXllJD16vxLBuRcmm3vg6U9lVaYpA'
+    - 'https://docs.google.com/spreadsheets/u/1/d/e/2PACX-1vQ7iYECa_ziPWygKBf9Og-PkQl5-gdnFRU-jP-z1FyjLfLXMAdRJN1J5Lx31NjnCq5lkRcL-T0Idh4_/pubhtml'
+    - 'https://docs.google.com/spreadsheets/d/e/2PACX-1vSbbciMU7t5dCB8auk-nyK-p-25TJjgUrzBuMCAiq_2ZDqoMSFAlZCctIC7y3TXS71ys_MRdoAU4Jqr/pubchart?oid=1729348495'
```
Comment (rxu) [inline collection/sites/googledocs.yml:6]: This example in not workable
image

File: collection/sites/googledocs.yml (hunk 1 above)
Comment (Tatiana5) [reply collection/sites/googledocs.yml:1]: Simultaneous enabling functions does not cause errors. Just links to documents, presentations and forms do not work

File: collection/sites/googledocs.yml (hunk 3 above)
Comment (Tatiana5) [reply collection/sites/googledocs.yml]: Fixed

**Review by iMattPro — CHANGES_REQUESTED** [review]
This breaks old posts using Google Sheets. Also the name and the wording to disable sheets is problematic and causes problems.

Please break it up into separate scripts. One for **Google Docs**, one for **Google Forms** and one for **Google Presentations**. Also include appropriate example links for each type of Doc.

**Comment by Tatiana5** [comment]
> Please break it up into separate scripts

I would really like to do this, but the code does not allow you to create more than one working file with the same 'host'

**Comment by iMattPro** [comment]
If that’s the case, please ask the author of the Media Embed plugin to add support for the other docs. We can't add this to the official extension like this, but you can post it here instead https://www.phpbb.com/customise/db/extension/mediaembed/support/topic/174656

The OneDrive we can keep here, however.

**Comment by iMattPro** [comment]
See https://github.com/s9e/TextFormatter/issues/190

**Comment by iMattPro** [comment]
Since this won't be addressed by the plugin author or us, please post your idea for this in the forum here instead https://www.phpbb.com/customise/db/extension/mediaembed/support/topic/174656

**Pull request closed without merging**